src/modules/adaptors/Convex/ConvexAdaptor.sol

```diff
@@ -0,0 +1,268 @@
+        _validatePositionIsUsed(pid, rewardsPool);
+        (address lpToken, , , , , ) = booster.poolInfo(pid);
+        ERC20 lpt = ERC20(lpToken); // TODO: double check that struct object is coming out of this properly
+        lpt.safeApprove(address(booster), assets);
+        booster.deposit(pid, assets, true);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(lpt, address(booster));
+    }
+
+    /**
+     * @notice If a user withdraw needs more LPTs than what is in the Cellar's wallet, then the Cellar will unstake, unwrap cvxLPTs, and withdraw LPTs from Convex
+     * @param amount of LPT to unstake, unwrap, and withdraw from Convex market
+     * @param receiver see baseAdaptor.sol
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdraw(
+        uint256 amount,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public override {
+        // Check that position is setup to be liquid.
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        (uint256 pid, address rewardPool) = abi.decode(adaptorData, (uint256, address));
+        _validatePositionIsUsed(pid, rewardPool);
+
+        //TODO: logic that checks if there is enough liquid curveLPT,and if not it does withdrawAndUnwrap(). If this logic is in place, withdrawableFrom() can report staked amount too.
+
+        // booster.withdraw(pid, amount);
+
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(rewardPool);
+        baseRewardPool.withdrawAndUnwrap(amount, true); // TODO: not sure if we just always set this to true (might be gas intensive), or if we allow this as a param somehow.
+    }
+
+    /**
+     * @notice Functions Cellars use to determine the withdrawable balance from an adaptor position.
+     * @dev Accounts for LPTs in the Cellar's wallet, and staked in Convex Market.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
+            (, address rewardPool) = abi.decode(adaptorData, (uint256, address));
+            IBaseRewardPool baseRewardPool = IBaseRewardPool(rewardPool);
+            ERC20 stakingToken = ERC20(baseRewardPool.stakingToken());
+            return (stakingToken.balanceOf(msg.sender) + baseRewardPool.balanceOf(msg.sender));
```

Comment:
The curve adaptor already checks for LP tokens sitting in the cellar wallet, so only report LP tokens staked on convex

Reply:
Makes sense, change made :)